ambry-frontend/src/main/java/com/github/ambry/frontend/NamedBlobPutHandler.java

```diff
@@ -632,18 +650,23 @@ private void updateVersionStateAndDeleteDatasetVersionOutOfRetentionCount() thro
                   NAMED_BLOB_PREFIX + SLASH + accountName + SLASH + containerName + SLASH + datasetName + SLASH
                       + version, requestPath.getSubResource(), requestPath.getBlobSegmentIdx());
           LOGGER.trace("New request path : " + newRequestPath);
-          // Replace RequestPath in the RestRequest and call DeleteBlobHandler.handle.
-          restRequest.setArg(InternalKeys.REQUEST_PATH, newRequestPath);
-          restRequest.setArg(InternalKeys.TARGET_ACCOUNT_KEY, null);
-          restRequest.setArg(InternalKeys.TARGET_CONTAINER_KEY, null);
-          //set the rest method to delete in order to delete data in named blob.
-          restRequest.setRestMethod(RestMethod.DELETE);
-          deleteBlobHandler.handle(restRequest, restResponseChannel,
-              recursiveCallback(datasetVersionRecordList, 1, accountName, containerName, datasetName));
+          // Replace RequestPath in the WrappedRestRequest for delete and call DeleteBlobHandler.handle.
```

Comment:
BTW, in line 643, could you also print out the account name, container name and the dataset name?

Reply:
Sure, Updated.